File: Week1/heroleggo/acmicpc-1181/solution.py
# 숫자 받기
N = int(input())

# 단어를 저장할 list
words = []

for i in range(N):
    words.append(input())

# list(set(list))를 통해 중복 제거하기
words = list(set(words))

# 길이, 사전순 배열을 조건 순서로 정렬
words.sort(key=lambda x: (len(x), x))

# 출력
for i in words:
    print(i)

File: Week1/heroleggo/acmicpc-1181/solution.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [length, ...data] = input;

const result = Array.from(new Set(data));

result.sort((a, b) => {
    if (a.length === b.length) {
        return a < b ? -1 : a > b ? 1 : 0;
    } else {
        return a.length - b.length;
    }
})

console.log(result.join('\n'));